h2o-py/h2o/explanation/_explain.py

```diff
@@ -726,13 +730,21 @@ def shap_summary_plot(
         col_name = top_n_features[i]
         col = contributions[permutation, col_name]
         dens = _density(col)
+        color = (
+            _uniformize(frame, col_name)[permutation]
+            if colorize_factors or not frame.isfactor(col_name)
+            else np.full(frame.nrow, 0.5)
+        )
+
+        if not np.any(np.isfinite(color)) or np.nanmin(color) == np.nanmax(color):
+            plt.scatter(0, i, alpha=alpha, c="grey")
+            continue  # constant variable; plotting it can throw StopIteration in some versions of matplotlib
+
         plt.scatter(
             col,
             i + dens * np.random.uniform(-jitter, jitter, size=len(col)),
             alpha=alpha,
-            c=_uniformize(frame, col_name)[permutation]
-            if colorize_factors or not frame.isfactor(col_name)
-            else np.full(frame.nrow, 0.5),
+            c=color,
             cmap=colormap
         )
```

Comment:
fixes https://github.com/h2oai/h2o-3/issues/15743

Reply:
the problem is a bug in matplotlib 3.7.1 and in later versions of matplotlib (3.7.2+) is fixed